### PR TITLE
Quick guided update execution for NK Storage

### PR DESCRIPTION
**Pull request by szszszsz**
Here is a quick guided update process execution for Nitrokey Storage. It simply executes commands from the documentation.
Marked as experimental due to:
- lack of testing for the availability of the used tool;
- using `sudo` (instead of installing Udev rules first).

Additionally added another potential library folder (for Fedora), and printing how to use `LIBNK_PATH` env var.

### CODE REVIEW

File: pynitrokey/cli/storage.py [hunk 1]
```diff
@@ -6,27 +6,85 @@
...
+    """experimental: run assisted update through dfu-programmer tool"""
+    if platform.system() != "Linux" or not experimental:
+        local_print("This feature is Linux only and experimental, which means it was not tested thoroughly.\n"
+                    "Please pass --experimental switch to force running it anyway.")
+        raise click.Abort()
+    assert firmware.endswith('.hex')
+
+    commands = f"""
+        sudo dfu-programmer at32uc3a3256s erase
+        sudo dfu-programmer at32uc3a3256s flash --suppress-bootloader-mem {firmware}
```
Comment (robin-nitrokey) [inline pynitrokey/cli/storage.py:44]: Couldn’t this cause issues if `firmware` contains a space?

**Review by robin-nitrokey — APPROVED** [review]
Not tested, but the code looks good.  I’d prefer if users were running `nitropy` with root privileges instead of calling `sudo` from `nitropy` (more explicit and more flexible).

File: pynitrokey/cli/storage.py (hunk 1 above)
Comment (szszszsz) [reply pynitrokey/cli/storage.py:44]: Indeed, good catch - to be corrected

**Comment by szszszsz** [comment]
> Not tested, but the code looks good. I’d prefer if users were running nitropy with root privileges instead of calling sudo from nitropy (more explicit and more flexible).

That would be my choice as well. The problem with this is, that user will have to have either multiple installations (local for root and user, or one system-wide), or refer to the user installation when calling, since user binaries are not within path while in `root`. We discourage running pynitrokey from root as well AFAIR. The default PIP installation now is user-local. 
Can we somehow work this around?

**Comment by szszszsz** [comment]
Ideally we can remove `sudo`, and add bootloader's VID PID to the Udev rule set - best option I could see so far.

**Comment by szszszsz** [comment]
- [ ] drop sudo use
- [ ] ask user to have the rules in place in case of connection issues
- [ ] check if device is available in either modes (application or bootloader)
- [ ] check for the used tool to be available 
- [ ] escape firmware path

File: pynitrokey/cli/storage.py (hunk 1 above)
Comment (szszszsz) [reply pynitrokey/cli/storage.py]: Corrected in latest update

**Comment by szszszsz** [comment]
Ready for review

**Comment by szszszsz** [comment]
Todo:
- [ ] Sign commits

<details>
 <summary>Example output </summary>

```
(venv) sz@stumpy ~/w/pynitrokey (sz-nkstorage-test)> env NITROPY_FIRMWARE_PASSWORD=12345678 NKDEV=1 nitropy storage upd
ate ~/work/nitrokey-storage-firmware/src/nitrokey-storage-V0.57.hex --experimental
Nitrokey tool for Nitrokey FIDO2, Nitrokey Start, Nitrokey 3 & NetHSM 0.4.19

* Running       "dfu-programmer --version"
Tool found: dfu-programmer 0.7.2

Note: During the execution update program will try to connect to the device. Check your udev rules in case of connection issues.
Using firmware path: /home/sz/work/nitrokey-storage-firmware/src/nitrokey-storage-V0.57.hex
Commands to be executed: 
        dfu-programmer at32uc3a3256s erase
        dfu-programmer at32uc3a3256s flash --suppress-bootloader-mem /home/sz/work/nitrokey-storage-firmware/src/nitrokey-storage-V0.57.hex
        dfu-programmer at32uc3a3256s start
        
Do you want to perform the firmware update now? [y/N]: y
Enabling firmware update mode
setting firmware update mode - success!
* Running       "dfu-programmer at32uc3a3256s erase"                                                                   
Checking memory from 0x2000 to 0x3FFFF...  Not blank at 0x2001.
Erasing flash...  Success
Checking memory from 0x2000 to 0x3FFFF...  Empty.

* Running       "dfu-programmer at32uc3a3256s flash --suppress-bootloader-mem $FIRMWARE"
Checking memory from 0x2000 to 0x3DDFF...  Empty.
0%                            100%  Programming 0x3BE00 bytes...
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>]  Success
0%                            100%  Reading 0x3E000 bytes...
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>]  Success
Validating...  Success
0x3BE00 bytes written into 0x3E000 bytes memory (96.57%).

* Running       "dfu-programmer at32uc3a3256s start"

Finished!
:: 'Nitrokey Storage' keys:                                                                                            
 - Nitrokey Storage 1-2.4.2:1.2
Found libnitrokey version: (3, 5)
Firmware version: (0, 57)
Admin PIN retries: 3
User PIN retries: 3


```

</details>

**Comment by szszszsz** [comment]
Pasting summary from the commit message:
```
- Add Nitrokey Storage update process support - simply executes the steps
from the documentation, decreasing potential chance of user's mistake.
    - Tested on Linux only. Requires passing --experimental flag on
      other OSes to execute.
    - Requires updated Udev rules, containing access rule for the
      Nitrokey Storage's bootloader (to be done).
- Support libnitrokey 3.7.0 if found.
    - Hint LIBNK_PATH env variable if none can be found.
- Ask users to report issues to Support Forum.
- Print stacktrace when NKDEV enviromental variable is set.

Tested:
    - Nitrokey Storage v0.56
    - dfu-programmer 0.7.2
    - libnitrokey 3.5.0

```

**Pull request closed without merging**